pygmt/tests/test_grdimage_image.py

```diff
@@ -39,7 +37,7 @@ def test_grdimage_image_dataarray(xr_image):
     Plot a 3-band RGB image using xarray.DataArray input.
     """
     fig = Figure()
-    fig.grdimage(grid=xr_image)
+    fig.grdimage(grid=xr_image, region="d")
```

Comment:
Need to set `region="d"` here, because `fig.grdimage(grid="@earth_day_01d")` plots using `projection="Q15c"` by default, whereas `fig.grdimage(grid=xr_image)` uses `projection="X15c/15c"`.

| Baseline (`@earth_day_01d`) | Result (xr_image) |
|--|--|
![baseline](https://github.com/user-attachments/assets/5c46af24-7faa-410a-bce0-00eefee831a7) | ![result](https://github.com/user-attachments/assets/60fbf5da-8fda-4fb2-9a5e-657b3344df6f) |

Reply:
Currently, we write the 3-band xarray.DataArray into a temporary GeoTIFF file. Is it because we don't set the CRS?

Reply:
Oh yes, good point! I've set the projection in commit 471c6e8b2364dbcbf9b3419133a62e8a6525796b, and `grdimage` now plots both of them the same way.